src/plugins/GUITestBase/src/tests/common_scenarios/Assembling/extract_consensus/GTTestsAssemblyExtractConsensus.cpp

```diff
@@ -0,0 +1,327 @@
+}
+
+GUI_TEST_CLASS_DEFINITION(ugenedb_test_0001) {
+    const GTLogTracer lt;
+    const QString path = testDir + "_common_data/ugenedb/scerevisiae.bam.ugenedb";
+
+    //  1. Select "Tools->NGS data analysis->Extract consensus from assemblies..."
+    //  2. Set "_common_data/ugenedb/scerevisiae.bam.ugenedb" as an input
+    //  3. Click "Run"
+    GTUtilsDialog::waitForDialog(os, new WizardFiller(
+        os, "Extract Consensus Wizard", new ExtractConsensusWizardScenario(QStringList() << path)
+    ));
+    GTMenu::clickMainMenuItem(os, QStringList() << "Tools"
+                                                << "NGS data analysis"
+                                                << "Extract consensus from assemblies...");
+
+    //  4. Wait for workflow finished
+    GTUtilsTaskTreeView::waitTaskFinished(os, 100000);
+    //  Expected state: There should be no errors in the log
+    CHECK_SET_ERR(!lt.hasErrors(), "Errors in log: " + lt.getJoinedErrorString());
+    //  There should be no notifications in the dashboard
+    CHECK_SET_ERR(!GTUtilsDashboard::hasNotifications(os), "Unexpected notification");
+
+    //  5. Return to workflow and call the Extract consensus wizard
+    GTUtilsWorkflowDesigner::returnToWorkflow(os);
+    //  6. Fill in the dialog :
+    //      {Algorithm} SAMtools
+    //      {Keep gaps} False
+    //      {Output file}
+    //     And click "Run"
+    GTUtilsDialog::waitForDialog(os, new WizardFiller(
+        os,
+        "Extract Consensus Wizard",
+        new ExtractConsensusWizardScenario(QStringList() << path, Algo::SAMTOOLS, false, "")
+    ));
+    GTToolbar::clickButtonByTooltipOnToolbar(os, MWTOOLBAR_ACTIVEMDI, "Show wizard");
+
+    //  7. Wait for workflow finished
+    GTUtilsTaskTreeView::waitTaskFinished(os, 100000);
+    //  Expected state: There should be no errors in the log
+    CHECK_SET_ERR(!lt.hasErrors(), "Errors in log: " + lt.getJoinedErrorString());
+    //  There should be no notifications in the dashboard
+    CHECK_SET_ERR(!GTUtilsDashboard::hasNotifications(os), "Unexpected notification");
+}
+
+GUI_TEST_CLASS_DEFINITION(multiple_inp_test_0001) {
+    struct ExtractConsensusWizardWithAddScenario : CustomScenario {
+        enum class Algorithm {
+            DEFAULT,
+            SAMTOOLS
+        };
+
+        QStringList inpPaths;
+        Algorithm algo;
+        bool keepGaps;
+        QString outFile;
+
+        ExtractConsensusWizardWithAddScenario(const QStringList &assembliesPath,
+                                              const Algorithm algo = Algorithm::DEFAULT,
+                                              const bool keepGaps = true,
+                                              const QString &outputFileName = "consensus.fa")
+            : inpPaths(assembliesPath), algo(algo), keepGaps(keepGaps), outFile(outputFileName) {
+        }
+
+        void run(HI::GUITestOpStatus &os) {
+            QWidget *const dialog = QApplication::activeModalWidget();
```

Comment:
nit: 
Use GTWidget::getActiveModalWidget.
this method 1) Waits 2) Asserts that the model widget exists.